src/test/java/com/connexta/store/clients/IndexClientImplTest.java

```diff
@@ -0,0 +1,98 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import com.connexta.store.WebClientTestUtils;
+import com.connexta.store.exceptions.common.DetailedResponseStatusException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.UUID;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@ExtendWith(MockitoExtension.class)
+class IndexClientImplTest {
+
+  private static final UUID DATASET_ID = UUID.randomUUID();
+  private static final URL FILE_URL;
+  private static final URL IRM_URL;
+  private static final URL METACARD_URL;
+
+  static {
+    try {
+      FILE_URL = new URL("http://localhost:9090/file");
+      IRM_URL = new URL("http://localhost:9090/irm");
+      METACARD_URL = new URL("http://localhost:9090/metacard");
+    } catch (MalformedURLException e) {
+      throw new RuntimeException();
+    }
+  }
+
+  private IndexClient indexClient;
+
+  @Test
+  void testIndexDataset() throws Exception {
+    // setup
+    ExchangeFunction exchangeFunction = mock(ExchangeFunction.class);
+    WebClient webClient =
+        WebClientTestUtils.mockEmptyResponseWebClient(HttpStatus.OK, exchangeFunction);
+
+    IndexClient indexClient = new IndexClientImpl(webClient);
+
+    // when
+    indexClient.indexDataset(DATASET_ID, FILE_URL, IRM_URL, METACARD_URL);
+
+    // then
+    ArgumentCaptor<ClientRequest> indexRequestCaptor = ArgumentCaptor.forClass(ClientRequest.class);
+    verify(exchangeFunction).exchange(indexRequestCaptor.capture());
+
+    ClientRequest indexRequest = indexRequestCaptor.getValue();
+    assertThat(indexRequest.method(), is(HttpMethod.PUT));
+    assertThat(indexRequest.url().toASCIIString(), is("/index/" + DATASET_ID));
+
+    JSONObject jsonObject = new JSONObject(WebClientTestUtils.getRequestBodyAsString(indexRequest));
+    assertThat(jsonObject.get("irmLocation"), is(IRM_URL.toString()));
+    assertThat(jsonObject.get("metacardLocation"), is(METACARD_URL.toString()));
+    assertThat(jsonObject.get("fileLocation"), is(FILE_URL.toString()));
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {400, 401, 403, 500, 501})
+  void testNonSuccessHttpResponse(int statusCode) {
+    // setup
+    ExchangeFunction exchangeFunction = mock(ExchangeFunction.class);
+    WebClient webClient =
+        WebClientTestUtils.mockEmptyResponseWebClient(
+            HttpStatus.valueOf(statusCode), exchangeFunction);
```

Comment:
✏️ 
```java
@ParameterizedTest
@EnumSource(
    value = HttpStatus.class,
    names = {"OK"},
    mode = EnumSource.Mode.EXCLUDE)
void testNonSuccessHttpResponse(HttpStatus httpStatus) {
  // setup
  ExchangeFunction exchangeFunction = mock(ExchangeFunction.class);
  WebClient webClient =
      WebClientTestUtils.mockEmptyResponseWebClient(
          httpStatus, exchangeFunction);
```

Reply:
I only want to handle what the API says it will return. Will this include every reponse code from the HttpStatus class excluding OK?

Reply:
Yes, that's how it would work. I'm fine only testing the exact status codes defined in the API.